dockerfiles/e2e-saas/docker-entrypoint.sh

```diff
@@ -59,13 +59,23 @@ length=${#USERNAME}
 
 echo "Trying to find token for $(echo $USERNAME | cut -c1-3) $(echo $USERNAME | cut -c4-$length)"
 
-#verify environment - if production or prod-preview
-#variable preview is used to differ between prod and prod-preview urls
-rm -rf cookie-file loginfile.html
-if [[ "$USERNAME" == *"preview"* ]] || [[ "$USERNAME" == *"saas"* ]]; then
-  preview="prod-preview."
-else
-  preview=""
+if [ -z $ACCOUNT_ENV ]; then
+  echo "ACCOUNT_ENV variable is not set, retrieving env from URL."
+
+  #verify environment - if production or prod-preview
+  #variable preview is used to differ between prod and prod-preview urls
+  rm -rf cookie-file loginfile.html
+  if [[ "$USERNAME" == *"preview"* ]] || [[ "$USERNAME" == *"saas"* ]]; then
+    preview="prod-preview."
+  else
+    preview=""
+  fi
+else 
+  if [ "$ACCOUNT_ENV" == "prod" ]; then
+    preview=""
+  else 
+    preview="prod-preview."
+  fi
 fi
```

Comment:
this logic is not straight forward for me. I believe we can do a better job of determining staging vs prod environment + would be nice to extract it to a separate function

Reply:
I can definitely separate that to a function. Do you have any idea how to improve this logic? I'm not aware of any other check.

Reply:
first `ACCOUNT_ENV variable is not set, retrieving env from URL` is it really true? according to the logic preview vs prod is retrieved based on the username?

Reply:
Yeah, it is retrieved from username. I'll fix that. Theoretically, we can remove that logic and just go with the mandatory variable `ACCOUNT_ENV` that would specify the env. But that would require changes in more scripts. If you want, I can do that as a part of this PR.